### PR TITLE
2way replay protection

### PR DESCRIPTION
**Pull request by leto**
This is an untested work in progress, I wanted to start a Pull Request to give some transparency and allow people to review.

### CODE REVIEW

**Comment by jl777** [comment]
maybe to be "compatible" with sbtc, using lower case for the signature concat string?
doesnt really matter, but having different cases for the different forks could lead to avoidable errors in future forks

**Comment by leto** [comment]
@jl777 sounds good, consistency is a nice thing to have

**Comment by nardeas** [comment]
Good job! So excited about this!

**Comment by airk42** [comment]
The build is broken now

./script/interpreter.h:92:43: error: left shift count >= width of type [-Werror=shift-count-overflow]
     SCRIPT_ENABLE_SIGHASH_FORKID = (1U << 42),
                                          
./script/interpreter.h:92:40: error: right operand of shift expression ‘(1u << 42)’ is >= than the precision of the left operand [-fpermissive]
     SCRIPT_ENABLE_SIGHASH_FORKID = (1U << 42),
                                    
./script/interpreter.h:92:45: error: enumerator value for ‘SCRIPT_ENABLE_SIGHASH_FORKID’ is not an integer constant
     SCRIPT_ENABLE_SIGHASH_FORKID = (1U << 42),
                                             

1U - is 4 bytes (32 bits) on all 3 OS's (linux, win, mac) in both 32 and 64 versions
so shifting it for 42 bits is an error

File: src/script/interpreter.cpp [hunk 1]
```diff
@@ -196,9 +196,26 @@ bool static CheckSignatureEncoding(const valtype &vchSig, unsigned int flags, Sc
...
+    if ((flags & SCRIPT_VERIFY_STRICTENC) != 0) {
+        if(!IsDefinedHashtypeSignature(vchSig)) {
+            return set_error(serror, SCRIPT_ERR_SIG_HASHTYPE);
+        }
+        unsigned char nHashType = vchSig[vchSig.size() - 1];
+        bool btcpForkHash       = nHashType & SIGHASH_FORKID;
+        bool btcpforkEnabled    = flags & SCRIPT_ENABLE_SIGHASH_FORKID;
+
+        if(!btcpForkHash && btcpforkEnabled) {
+            return set_error(serror, SCRIPT_ERR_ILLEGAL_BTCP_FORKID);
```
Comment (airk42) [inline src/script/interpreter.cpp]: SCRIPT_ERR_ILLEGAL_BTCP_FORKID was never defined

**Pull request merged**